app/frontend/tokenRegistry/tokenRegistryApi.ts

```diff
@@ -0,0 +1,37 @@
+import request from '../wallet/helpers/request'
+import {ADALITE_CONFIG} from '../config'
+import {RegisteredTokenMetadata} from '../types'
+import {TokenRegistryApi} from './types'
+
+export default (): TokenRegistryApi => {
+  const getTokensMetadata = async (
+    subjects: string[]
+  ): Promise<{[subject: string]: RegisteredTokenMetadata}> => {
+    const url = `${ADALITE_CONFIG.ADALITE_SERVER_URL}/api/tokenRegistry/getTokensMetadata`
+    const requestBody = {subjects}
+    const response = await request(url, 'POST', JSON.stringify(requestBody), {
+      'Content-Type': 'application/json',
+    })
+    if (response?.Right) {
+      return response.Right.reduce((acc, tokenMetadata) => {
```

Comment:
I'd split this into the "fetch" and "parse" part and add some basic unit test for parsing so we have actual examples of how the data returned from the server vs the representation in our code looks like, which could serve as a valuable reference when trying to understand what's going on

Reply:
Refactored and added test